src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/entities/ExpCollector.java

```diff
@@ -44,35 +44,38 @@ public class ExpCollector extends SlimefunItem implements InventoryBlock, Energy
 
     private final int[] border = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 };
 
-    private static final int ENERGY_CONSUMPTION = 10;
     private static final String DATA_KEY = "stored-exp";
 
+    private final double range;
+    private int energyConsumedPerTick = -1;
+    private int energyCapacity = -1;
+
     @ParametersAreNonnullByDefault
-    public ExpCollector(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+    public ExpCollector(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe, double range) {
         super(itemGroup, item, recipeType, recipe);
 
         createPreset(this, this::constructMenu);
 
+        this.range = range;
```

Comment:
any reason this is done below createPreset?
this. in constructor should always be at the top